server/src/main/java/keywhiz/service/resources/automation/v2/BackfillRowHmacResource.java

```diff
@@ -38,11 +50,65 @@
 
   private final DSLContext jooq;
   private final RowHmacGenerator rowHmacGenerator;
+  private final AuditLog auditLog;
 
   @Inject
-  public BackfillRowHmacResource(DSLContext jooq, RowHmacGenerator rowHmacGenerator) {
+  public BackfillRowHmacResource(
+      DSLContext jooq,
+      RowHmacGenerator rowHmacGenerator,
+      AuditLog auditLog) {
     this.jooq = jooq;
     this.rowHmacGenerator = rowHmacGenerator;
+    this.auditLog = auditLog;
+  }
+
+  /**
+   * Backfill row_hmac for a specific secret.
+   */
+  @Timed @ExceptionMetered
+  @Path("secret/name/{secretName}")
+  @POST
+  @Consumes(APPLICATION_JSON)
+  @Produces(APPLICATION_JSON)
+  public void backfillSecretRowHmacByName(
+      @Auth AutomationClient automationClient,
+      @PathParam("secretName") String secretName,
+      @QueryParam("force") boolean force) {
+    Optional<SecretsRecord> maybeRow = jooq.selectFrom(SECRETS)
+        .where(SECRETS.NAME.eq(secretName))
+        .fetchOptional();
```

Comment:
Originally I was thinking that authentication was handled at the class level, but it's not: I've now added @Auth AutomationClient to enforce that this is an authenticated, high-privilege caller (with tests to verify).